build_web_compilers/lib/src/ddc_names.dart

```diff
@@ -2,10 +2,25 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// Escape [name] to make it into a valid identifier.
+import 'package:path/path.dart' as p;
+
+/// Transforms a path to a valid JS identifier.
 ///
-/// This is copied from
-/// https://github.com/dart-lang/sdk/blob/300fd66f5c5e7497674ea4dd9fd21be91ee3c71e/pkg/dev_compiler/lib/src/analyzer/code_generator.dart#L6390
+/// This logic must be synchronized with [pathToJSIdentifier] in DDC at:
```

Comment:
How synchronized exactly? How much backwards and forwards compatibility do we have?

Reply:
It should be logically identical (so copy-paste). Luckily, pathToJSIdentifier won't change much.

Reply:
We need to document in the commit message exactly what the rollout looks like. Something like.

"An upcoming release of the SDK will break compatibility with older versions of `build_web_compilers` that use the existing pattern. This change makes the compilers forwards compatible with upcoming SDK and remains compatible with old SDKs."

Then this sentence in the doc should be made more clear. Something like

"If the SDK changes patterns this logic should be updated to accept the new pattern in addition to the old pattern."

We can't be "synchronized" with the SDK because we don't have 1:1 releases with the SDK.

Reply:
An ideal case would be to use some standard Uri to JS-compliant identifier scheme that could be shared across our codebase. Synchronicity has been an implicit requirement since DDC decided to export names in a special way. However, this is a pretty unique case, as the scheme usually isn't volatile.

Reply:
As Nate mentioned synchronicity just isn't possible though. There is an issue filed somewhere about getting a better api from ddc that would allow us to not have to hardcode so much boostrapping logic so we wouldn't own this type of thing at all, which is really the only way forward.